### PR TITLE
New backoffice/add system text json configuration attribute

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- Added new ManagementApiBaseController that all of the api in ManagementApi project should inherit from
- Created new EnumOutput formatter that now adds the JsonStringEnumConverter to our converts, so we serialize enum by text and not by number.
- Added new SystemTextJsonConfigurationAttribute, that now ensures our properties gets serialized by camelCase so we don't have to explictly set `JsonPropertyName("name")` on every property. and ofcourse also constaints the new EnumOutputFormatter

# How to test
- I just copied the GetAnalyticsController so i could test it, either do that or make your own endpoint that returns and enum
- Call the endpoint with via postman GET: https://localhost:44331/umbraco/api/v1.0/analytics 
- Assert that the property comes out as camelCase, IE: `analyticsLevel` and the value is now text, IE: `Basic`
```
using Microsoft.AspNetCore.Mvc;
using NSwag.Annotations;
using Umbraco.New.Cms.Web.Common.Routing;

namespace Umbraco.Cms.ManagementApi.Controllers.Analytics;

[ApiController]
[BackOfficeRoute("api/v{version:apiVersion}/analytics")]
[OpenApiTag("Analytics")]
public abstract class AnalyticsControllerBase : MananagementApiBaseController
{
}

```
```
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Umbraco.Cms.Core.Services;
using Umbraco.Cms.ManagementApi.ViewModels.Analytics;

namespace Umbraco.Cms.ManagementApi.Controllers.Analytics;

[ApiVersion("1.0")]
public class GetAnalyticsController : AnalyticsControllerBase
{
    private readonly IMetricsConsentService _metricsConsentService;

    public GetAnalyticsController(IMetricsConsentService metricsConsentService) => _metricsConsentService = metricsConsentService;

    [HttpGet]
    [MapToApiVersion("1.0")]
    [ProducesResponseType(typeof(AnalyticsLevelViewModel), StatusCodes.Status200OK)]
    public async Task<AnalyticsLevelViewModel> Get() => new() { AnalyticsLevel = _metricsConsentService.GetConsentLevel() };
}

```
View model we use:
```
using Umbraco.Cms.Core.Models;

namespace Umbraco.Cms.ManagementApi.ViewModels.Analytics;

public class AnalyticsLevelViewModel
{
    public TelemetryLevel AnalyticsLevel { get; set; }
}

```

### CODE REVIEW

File: src/Umbraco.Cms.ManagementApi/Controllers/MananagementApiBaseController.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.ManagementApi.Filters;
+
+namespace Umbraco.Cms.ManagementApi.Controllers;
+
+[SystemTextJsonConfiguration]
+public class MananagementApiBaseController : Controller
```
Comment (kjac) [inline src/Umbraco.Cms.ManagementApi/Controllers/MananagementApiBaseController.cs:7]: Suggest renaming to `MananagementApiControllerBase` to follow the general naming convention

File: src/Umbraco.Cms.ManagementApi/Filters/EnumOutputFormatter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Umbraco.Cms.ManagementApi.Filters;
+
+public class EnumOutputFormatter : SystemTextJsonOutputFormatter
```
Comment (kjac) [inline src/Umbraco.Cms.ManagementApi/Filters/EnumOutputFormatter.cs:7]: Suggest renaming to `ManagementApiJsonOutputFormatter` - we may end up doing more here than enum formatting.

File: src/Umbraco.Cms.ManagementApi/Filters/SystemTextJsonConfigurationAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Umbraco.Cms.ManagementApi.Filters;
+
+public class SystemTextJsonConfigurationAttribute : TypeFilterAttribute
```
Comment (kjac) [inline src/Umbraco.Cms.ManagementApi/Filters/SystemTextJsonConfigurationAttribute.cs:7]: Suggest renaming this to `ManagementApiJsonConfigurationAttribute`, to make it abundantly clear what it is used for.

**Review by kjac — CHANGES_REQUESTED** [review]
@Zeegaan the code works like a charm 😄 I have suggested some naming changes, have a look if you agree with them.

We must remember to apply this new base controller to all existing new-backoffice controllers. I will make a note of this 👍

**Comment by Zeegaan** [comment]
Pipelines failing has to do with the dev branch failing, not the latest change 👍

**Comment by Zeegaan** [comment]
@kjac I will go through and update the existing controllers in dev and in my PRs when this is merged aswell 🚀

**Review by kjac — APPROVED** [review]

**Pull request merged**